### PR TITLE
add benchmark for block_store

### PR DESCRIPTION
**Pull request by arvidn**
This patch adds a benchmark for `BlockStore`. For now, just `add_full_block()`. The time it takes to generate a fake full block is not included in the timings, this is intended to (primarily) be an I/O benchmark.

This patch also factors out some common parts from the `coin_store.py` benchmark.

tested on a fast computer on SSD:

```
45.1068s, add_full_blocks
all tests completed in 45.1068s
database size: 62.534 MB
```

tested on a spinning disk:

```
260.6673s, add_full_blocks
all tests completed in 260.6673s
database size: 62.370 MB
```

### CODE REVIEW

File: benchmarks/block_store.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+
+NUM_ITERS = 20000
+
+# we need seeded random, to have reproducible benchmark runs
+random.seed(123456789)
+
+
+def rand_class_group_element() -> ClassgroupElement:
+    return ClassgroupElement(secrets.token_bytes(100))
```
Comment (richardkiss) [inline benchmarks/block_store.py:31]: I'd recommend NOT using `secrets` here to maximize reproducibility.

File: benchmarks/block_store.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+            has_pool_pk = random.randint(0, 1)
+
+            proof_of_space = ProofOfSpace(
+                rand_hash(),  # challenge
+                rand_g1() if has_pool_pk else None,
+                rand_hash() if not has_pool_pk else None,
+                rand_g1(),  # plot_public_key
+                32,
+                secrets.token_bytes(8 * 32),
```
Comment (richardkiss) [inline benchmarks/block_store.py:129]:
```suggestion
                random.randbits(8 * 32).to_bytes(32, "big"), # not tested, but you get the idea
```

Comment (arvidn) [reply benchmarks/block_store.py:129]: this won't reliably produce the desired number of bytes though.. I'll come up with something else

**Comment by arvidn** [comment]
@richardkiss fixed

**Review by richardkiss — APPROVED** [review]
Great, let's merge it.

**Comment by wjblanke** [comment]
nice thanks guys

**Pull request merged**